tests/components/abode/common.py

```diff
@@ -0,0 +1,25 @@
+"""Common methods used across tests for Abode."""
+from unittest.mock import patch
+
+from homeassistant.components.abode import DOMAIN as ABODE_DOMAIN
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.setup import async_setup_component
+
+from tests.common import MockConfigEntry
+
+
+async def setup_platform(hass, platform):
+    """Set up the Abode platform."""
+    mock_entry = MockConfigEntry(
+        domain=ABODE_DOMAIN,
+        data={CONF_USERNAME: "user@email.com", CONF_PASSWORD: "password"},
+    )
+    mock_entry.add_to_hass(hass)
+
+    with patch("homeassistant.components.abode.ABODE_PLATFORMS", [platform]), patch(
+        "abodepy.event_controller.sio"
+    ), patch("abodepy.utils.save_cache"):
+        assert await async_setup_component(hass, ABODE_DOMAIN, {})
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5184102)